core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -165,6 +191,40 @@ LocalExchangeSource getSource(int partitionIndex)
         return sources.get(partitionIndex);
     }
 
+    private static PartitionFunction createPartitionFunction(
+            NodePartitioningManager nodePartitioningManager,
+            Session session,
+            BlockTypeOperators blockTypeOperators,
+            PartitioningHandle partitioning,
+            int partitionCount,
+            List<Type> partitionChannelTypes,
+            boolean isHashPrecomputed)
+    {
+        if (partitioning.getConnectorHandle() instanceof SystemPartitioningHandle) {
+            HashGenerator hashGenerator;
+            if (isHashPrecomputed) {
+                hashGenerator = new PrecomputedHashGenerator(0);
+            }
+            else {
+                hashGenerator = new InterpretedHashGenerator(partitionChannelTypes, IntStream.range(0, partitionChannelTypes.size()).toArray(), blockTypeOperators);
+            }
+            return new LocalPartitionGenerator(hashGenerator, partitionCount);
+        }
+
+        ConnectorBucketNodeMap connectorBucketNodeMap = nodePartitioningManager.getConnectorBucketNodeMap(session, partitioning);
```

Comment:
in

> Would it work if -- instead of using getConnectorBucketNodeMap().getBucketCount(), we used partitionCount here?

i mean just remove call to `getConnectorBucketNodeMap`, while still leave hash re-distribution here.